internal/backends/sqlite/metadata/registry.go

```diff
@@ -17,22 +17,35 @@ package metadata
 
 import (
 	"context"
-	"crypto/sha1"
 	"database/sql"
 	"encoding/hex"
 	"errors"
 	"fmt"
+	"hash/fnv"
+	"strings"
 
 	"go.uber.org/zap"
 	"modernc.org/sqlite"
 	sqlitelib "modernc.org/sqlite/lib"
 
 	"github.com/FerretDB/FerretDB/internal/backends/sqlite/metadata/pool"
 	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/must"
 )
 
-// metadataTableName is a SQLite table name where FerretDB metadata is stored.
-const metadataTableName = "_ferretdb_collections"
+const (
+	// Reserved prefix for database and collection names.
+	reservedPrefix = "_ferretdb_"
+
+	// Reserved prefix for SQLite table name.
+	reservedTablePrefix = "sqlite"
+
+	// SQLite table name where FerretDB metadata is stored.
+	metadataTableName = reservedPrefix + "collections"
+)
+
+// ErrInvalidCollectionName indicates that a collection didn't pass name checks.
+var ErrInvalidCollectionName = fmt.Errorf("invalid FerretDB collection name")
```

Comment:
Let's not introduce multiple error-handling ways and use `backends.Error` together with `backends.ErrorCodeIs`. Specifically, we already have `backends.ErrorCodeCollectionNameIsInvalid`